### PR TITLE
west status: only print output for projects with nonempty status

### PR DESCRIPTION
**Pull request by mbolivar-nordic**
The 'west status' output is no longer useful with many projects in the
mix. Apply a similar cleanup to 'west status' that 'west diff' got in
a53ec10cf28b4e500b5095eec33086cac1a6d839 ("west diff: only print
output for projects with nonempty diffs").

Signed-off-by: Martí Bolívar <marti.bolivar@nordicsemi.no>

### CODE REVIEW

**Comment by mbolivar-nordic** [comment]
cc @sjg20

**Comment by marc-hb** [comment]
> The 'west status' output is no longer useful with many projects in the mix

What I would prefer is some kind of `west update --if-present` option because not cloning at all the 95% of the projects I don't use is an older and also pretty good way to solve this problem

**Review by carlescufi — APPROVED** [review]
Looks good!

**Comment by mbolivar-nordic** [comment]
> What I would prefer is some kind of `west update --if-present` option because not cloning at all the 95% of the projects I don't use is an older and also pretty good way to solve this problem

Sounds like a nice future extension -- feel free to file an issue. I think that since `west status` already ignores inactive projects by default, and the upstream zephyr manifest now has groups, you may have the ingredients for most of what you want already by just cloning a subset of active projects.

**Comment by marc-hb** [comment]
> Sounds like a nice future extension -- feel free to file an issue.

Done in #519

File: src/west/app/project.py [hunk 1]
```diff
@@ -697,13 +697,77 @@ def do_run(self, args, user_args):
...
+                                 stderr=subprocess.PIPE,
+                                 cwd=project.abspath)
+
+        def has_output():
+            # 'git status --porcelain' prints nothing if there
+            # are no notable changes, so any output at all
+            # means we should run 'git status' on the project.
+            stdout, stderr = None, None
+            try:
+                stdout, stderr = popen.communicate(timeout=0.1)
```
Comment (marc-hb) [inline src/west/app/project.py:748]: ~This is racy~, new bug filed:
- #981

**Pull request merged**